### PR TITLE
[core] Add runtime API for setting tile prefetch delta for a Source

### PR DESCRIPTION
**Pull request by alexshalamov**
The new `Source::setPrefetchZoomDelta(optional<uint8_t> delta)` method allows overriding default tile prefetch setting that is defined by the Map instance. The method can be moved to a generic style specification if found to be useful for gl-js engine.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [x] tagged `@mapbox/maps-android @mapbox/maps-ios @mapbox/core-sdk` if this PR adds or updates a public API
 - [x] tagged `@mapbox/gl-js` if this PR includes shader changes or needs a js port
 - [x] apply `needs changelog` label if a changelog is needed (remove label when added)

/cc @mapbox/gl-js @mapbox/core-sdk

### CODE REVIEW

File: src/mbgl/annotation/annotation_source.cpp [hunk 1]
```diff
@@ -26,4 +30,8 @@ bool AnnotationSource::supportsLayerType(const mbgl::style::LayerTypeInfo* info)
     return !std::strcmp(info->type, "line") || !std::strcmp(info->type, "symbol") || !std::strcmp(info->type, "fill");
 }
 
+Mutable<Source::Impl> AnnotationSource::createMutable() const noexcept {
+    return staticMutableCast<Source::Impl>(makeMutable<Impl>(impl()));
```
Comment (pozdnyakov) [inline src/mbgl/annotation/annotation_source.cpp]: does it need explicit `staticMutableCast<Source::Impl>` ?

File: src/mbgl/renderer/render_orchestrator.cpp [hunk 2]
```diff
@@ -337,6 +337,11 @@ std::unique_ptr<RenderTree> RenderOrchestrator::createRenderTree(
                 renderItemsEmplaceHint = layerRenderItems.emplace_hint(renderItemsEmplaceHint, layer, nullptr, index);
             }
         }
+
+        if (auto sourcePrefetchDelta = sourceImpl->getPrefetchZoomDelta()) {
```
Comment (pozdnyakov) [inline src/mbgl/renderer/render_orchestrator.cpp]: Could this code be inside Render::SourceUpdate?

File: src/mbgl/annotation/annotation_source.cpp (hunk 1 above)
Comment (alexshalamov) [reply src/mbgl/annotation/annotation_source.cpp]: Yes, it needs :disappointed:

File: src/mbgl/renderer/render_orchestrator.cpp (hunk 2 above)
Comment (alexshalamov) [reply src/mbgl/renderer/render_orchestrator.cpp]: Update is implemented by concrete classes, I pushed implementation down to rendersource::update and added parameter to pyramid update. https://github.com/mapbox/mapbox-gl-native/pull/16179/commits/c7ab18af52087f1e483f91c4037248429460643a

**Comment by tobrun** [comment]
@alexshalamov downstream we are implementing code generation for Sources similar to what we already have for layers/properties/light etc from the style spec today. Should this configuration be exposed through the style spec? https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/

**Comment by alexshalamov** [comment]
@asheemmamoowala Do you think adding prefetchDelta to sources specification would be useful to gl-js? If so, I can start working on it next sprint.

@tobrun meanwhile, would it be possible to add an exception and generate setter / getter for all sources? I think generator already has exceptions, since some of the features are not implemented in native (e.g., [promoteId](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#vector-promoteId)).

**Comment by asheemmamoowala** [comment]
> @asheemmamoowala Do you think adding prefetchDelta to sources specification would be useful to gl-js? If so, I can start working on it next sprint.

@alexshalamov I'm going to have to defer to @kkaefer @mourner on this. Cancelling and managing tasks in WebWorkers is different from native threads, so this may not work as well in gl-js as it does in native.

**Pull request merged**